pkg/datamover/testutils.go

```diff
@@ -0,0 +1,438 @@
+	"fmt"
+	"math/big"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/kubernetes"
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	awsconfig "github.com/kanisterio/kanister/pkg/aws"
+	"github.com/kanisterio/kanister/pkg/controllers/repositoryserver"
+	"github.com/kanisterio/kanister/pkg/format"
+	"github.com/kanisterio/kanister/pkg/kopia"
+	kopiacmd "github.com/kanisterio/kanister/pkg/kopia/command"
+	"github.com/kanisterio/kanister/pkg/kopia/repository"
+	"github.com/kanisterio/kanister/pkg/kube"
+)
+
+const (
+	repositoryServerTestNamespace    = "repository-server-test-namespace-"
+	repositoryServerTestPod          = "repository-server-test-pod-"
+	repositoryServerTestService      = "repository-server-test-service-"
+	kanisterToolsImage               = "ghcr.io/kanisterio/kanister-tools:0.93.0"
+	kanisterToolsImageEnvName        = "KANISTER_TOOLS"
+	testAwsAccessKeyId               = "AKIAIOSFODNN7EXAMPLE"
+	testAwsAccessSecretKey           = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+	testAwsRegion                    = "us-west-2"
+	testAwsLocationEndpoint          = "http://minio.minio.svc.cluster.local:9000"
+	testAwsS3BucketName              = "tests.kanister.io"
+	defaultKopiaRepositoryPassword   = "test1234"
+	defaultKopiaRepositoryUser       = "repositoryuser"
+	defaultKopiaRepositoryPath       = "repository-server-test"
+	defaultKopiaRepositoryHost       = "localhost"
+	defaultServerStartTimeout        = 10 * time.Minute
+	testRepoServerName               = "test-repo-server"
+	testKopiaRepoServerAdminUsername = "testadmin@localhost"
+	testKopiaRepoServerUsername      = "testuser"
+	testKopiaRepoServerAdminPassword = "testpass1234"
+	testTLSKeyPath                   = "/tmp/tls/tls.key"
+	testTLSCertPath                  = "/tmp/tls/tls.crt"
+)
+
+func createRepositoryServerTestNamespace(ctx context.Context, cli kubernetes.Interface) (*corev1.Namespace, error) {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: repositoryServerTestNamespace,
+		},
+	}
+	nsCreated, err := cli.CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return nsCreated, nil
+}
+
+func getKanisterToolsImage() string {
+	if val, ok := os.LookupEnv(kanisterToolsImageEnvName); ok {
+		return val
+	}
+	return kanisterToolsImage
+}
+
+func createRepositoryServerTestPod(ctx context.Context, cli kubernetes.Interface, namespace string, secret *corev1.Secret) (*corev1.Pod, error) {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: repositoryServerTestPod,
+			Namespace:    namespace,
+			Labels:       map[string]string{"app": "test-repo-server"},
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "test-repository-server-container",
```

Comment:
Lets generate name for the pod over here, if the test fails and pod already exists, we will have to delete the pod manually before running the test again